src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -756,4 +948,156 @@ private double divideOrReturnZero(double numerator, int denominator) {
         return (denominator == 0) ? 0 : numerator / denominator;
     }
 
+    /**
+     * Calculate the Per Recipient Statistics for MSQ questions.
+     */
+    private static class MsqPerRecipientStatistics {
+        List<String> msqChoices;
+        boolean otherEnabled;
+        McqStatistics msqStats;
+
+        MsqPerRecipientStatistics(FeedbackMsqQuestionDetails msqDetails,
+                McqStatistics msqStats) {
+            this.msqChoices = msqDetails.getMsqChoices();
+            this.otherEnabled = msqDetails.getOtherEnabled();
+            this.msqStats = msqStats;
+        }
+
+        /**
+         * Calculates the number of responses for each response attribute.
+         */
+        private Map<String, Integer> getNumberOfResponsesForEachResponse(FeedbackMsqResponseDetails responseDetails,
+                Map<String, Integer> responseCountPerOption) {
+            List<String> answerStrings = responseDetails.getAnswerStrings();
+            boolean isOtherOptionAnswer = responseDetails.isOtherOptionAnswer();
+            String otherAnswer = "";
+
+            if (isOtherOptionAnswer) {
+                responseCountPerOption.put("Other", responseCountPerOption.get("Other") + 1);
+
+                // remove other answer temporarily to calculate stats for other options
+                otherAnswer = answerStrings.get(answerStrings.size() - 1);
+                answerStrings.remove(otherAnswer);
```

Comment:
Here I am removing the `Other` answer from answer strings, I need to store the value somewhere so that I can add it again after calculating response count for other options.